components/tidb_query/src/codec/datum.rs

```diff
@@ -192,7 +192,7 @@ impl Datum {
                 Ok(d.cmp(&d2))
             }
             _ => {
-                let f = bs.convert(ctx)?;
+                let f: f64 = bs.convert(ctx)?;
```

Comment:
Do we need declare type `f64` here? If need, please add `f64` to every case like this.

Reply:
Here we need, if in somewhere, it is necessary but we forget to add, rustc will report an error.